cocotb/share/lib/simulator/simulatormodule.h

```diff
@@ -73,68 +73,83 @@ static PyMethodDef SimulatorMethods[] = {
+        "register_timed_callback(time: int, func: Callable[..., None], *args: Any) -> cocotb.simulator.gpi_cb_hdl\n"
+        "Register a timed callback."
     )},
     {"register_value_change_callback", register_value_change_callback, METH_VARARGS, PyDoc_STR(
         "register_value_change_callback(signal, func, edge, /, *args)\n"
         "--\n\n"
-        "Register a signal change callback"
+        "register_value_change_callback(signal: cocotb.simulator.gpi_sim_hdl, func: Callable[..., None], edge: int, *args: Any) -> cocotb.simulator.gpi_cb_hdl\n"
+        "Register a signal change callback."
     )},
     {"register_readonly_callback", register_readonly_callback, METH_VARARGS, PyDoc_STR(
         "register_readonly_callback(func, /, *args)\n"
         "--\n\n"
-        "Register a callback for the read-only section"
+        "register_readonly_callback(func: Callable[..., None], *args: Any) -> cocotb.simulator.gpi_cb_hdl\n"
+        "Register a callback for the read-only section."
     )},
     {"register_nextstep_callback", register_nextstep_callback, METH_VARARGS, PyDoc_STR(
         "register_nextstep_callback(func, /, *args)\n"
         "--\n\n"
-        "Register a callback for the NextSimTime callback"
+        "register_nextstep_callback(func: Callable[..., None], *args: Any) -> cocotb.simulator.gpi_cb_hdl\n"
+        "Register a callback for the NextSimTime callback."
     )},
     {"register_rwsynch_callback", register_rwsynch_callback, METH_VARARGS, PyDoc_STR(
         "register_rwsynch_callback(func, /, *args)\n"
         "--\n\n"
-        "Register a callback for the read-write section"
+        "register_rwsynch_callback(func: Callable[..., None], *args: Any) -> cocotb.simulator.gpi_cb_hdl\n"
+        "Register a callback for the read-write section."
     )},
     {"stop_simulator", stop_simulator, METH_VARARGS, PyDoc_STR(
         "stop_simulator()\n"
         "--\n\n"
-        "Instruct the attached simulator to stop"
+        "stop_simulator() -> None\n"
+        "Instruct the attached simulator to stop. Users should not call this function."
     )},
     {"log_level", log_level, METH_VARARGS, PyDoc_STR(
         "log_level(level, /)\n"
         "--\n\n"
-        "Set the log level for GPI"
+        "log_level(level: int) -> None\n"
+        "Set the log level for GPI."
     )},
 
     {"get_sim_time", get_sim_time, METH_NOARGS, PyDoc_STR(
         "get_sim_time()\n"
         "--\n\n"
-        "Get the current simulation time as an int tuple"
+        "get_sim_time() -> Tuple[int, int]\n"
+        "Get the current simulation time.\n"
+        "\n"
+        "Time is represented as a tuple of 32 bit integers ([low32, high32]) comprising a single 64 bit integer."
     )},
     {"get_precision", get_precision, METH_NOARGS, PyDoc_STR(
         "get_precision()\n"
         "--\n\n"
-        "Get the precision of the simulator"
+        "get_precision() -> int\n"
+        "Get the precision of the simulator."
```

Comment:
Maybe this should be more specific, or have an example.

Reply:
Is what is now here clear enough?

Reply:
Yes, thanks!